spinalcordtoolbox/reports/qc2.py

```diff
@@ -114,13 +114,30 @@ def create_qc_entry(
         if not path.exists():
             raise FileNotFoundError(f"Required QC image '{img_type}' was not found at the expected path: '{path}'")
 
-    # Use mutex to ensure that we're only generating shared QC assets using one process at a time
-    realpath = path_qc.resolve()
-    with mutex(f"sct_qc-{realpath.name}-{md5(str(realpath).encode('utf-8')).hexdigest()}"):
-        # Create a json file for the new QC report entry
-        path_json = path_qc / '_json'
+    # Define where the JSON entry should be saved
+    path_json = path_qc / '_json'
+    path_result = path_json / f'qc_{timestamp}.json'
+
+    # Create a mutex for the QC file to avoid a race condition if multiple files try to write simultaneously
+    # KO, MB, and JN: The hash here is required to sanitize the path name. Otherwise, special characters for the OS
```

Comment:
Thanks for making this design constraint explicit in a code comment.

Also, tiny typo:

```suggestion
    # KO, MGP, and JN: The hash here is required to sanitize the path name. Otherwise, special characters for the OS
```

Reply:
Oh, whoops; thanks for catching that.